### ISSUE TITLE
RoomEvent.ActiveSpeakersChange Speakers list not right

### ISSUE THREAD

**Issue by shoyu666** [livekit/livekit#2228]
**Describe the bug**
Speakers list include mute user on the top。

**Server**
- Version: [[v1.5.0](https://github.com/livekit/livekit/releases/tag/v1.5.0)]
- Environment: [windows,mac,linux(dev)] same

**Client**
Device Info:
Device: [一加 Ace 2V] It seems to have nothing to do with the device
OS: [Android 13]
LiveKit SDK version: [1.2.1]

**To Reproduce**
1. device A,B,C join room （no one was mute）,device A speak,Speakers list return {A} 。
2. device A mute self immediately，device B speak。 Speakers list always return {A,B} 。
3. unmute A ，A speak，then B speak ,Speakers list return {B,A} 。
4. See error the problem is： step2,when A is mute Speakers list include A on the top。

**Expected behavior**
Speakers list not include mute user on the top

**Additional context**
when mute A，android sdk  JavaAudioDeviceModule.AudioRecordStateCallback No more callbacks in device A。

**Screenshot**
user "f064@661-137
e-4fab-9128-e" is mute but Blue border does not disappear
<img src="https://github.com/livekit/client-sdk-android/assets/20290812/2e2dc81f-ade0-4d6a-a8d1-8dd503347ca8" width="300">

**Comment by davidzhao** [comment]
Do not file dupe issues please.

Duplicate of https://github.com/livekit/client-sdk-android/issues/311

**Comment by davidliu** [comment]
@davidzhao This seems like an actual issue? The extra participants are irrelevant to the core issue, so it boils down to this:

1. Speak into mic: Receive speakers_changed with speaking = true.
2. Mute mic *while speaking*: No extra speakers_changed event is received to clear the speaking flag.

It's only after unmuting does the server seem to update the speaking status. Guessing the speaking calculations on server-side only fire when the media stream track is sending bits?

**Comment by boks1971** [comment]
Server should update on mute I believe. Has this started happening recently?

**Comment by davidliu** [comment]
Not sure, never paid attention to this case before. Reproed with Android client on cloud. 

Might be a specific interaction with native clients or the android sdk specifically, as I couldn't repro on JS.

**Comment by shoyu666** [comment]
update：
1:try unmute A quickly during A speak。
2: Mute is a way to trigger，and another way:
i hava a hdmi to usb device（uvc camera with audio），it can control audio data open or close。
when close send audio data(be different from mute by sdk),it happend。

device be similar to pic
<img src="https://github.com/livekit/livekit/assets/20290812/3dcbf9d9-3642-4815-a999-7a0c36eb00e8" width="300">

**Comment by shoyu666** [comment]
in android must mute  **before** Blue border  disappear

1: js works fine
2:use livekit-android sample-app can case android and js error(both)
<img width="1508" alt="image" src="https://github.com/livekit/livekit/assets/20290812/529bbebe-87ae-4a45-a9af-77e0a4600db0">

**Comment by shoyu666** [comment]
server act difference js mute and android-mute
js mute: changedSpeakers  { Active>>[true,true,....,false];}
android-mute:changedSpeakers  { Active>>[true,true,....,true];} with out false
https://github.com/livekit/livekit/blob/302facc60dae494f04ec2a94933b9e8a0379427e/pkg/rtc/room.go#L1214C10-L1214C25

**Comment by shoyu666** [comment]
I tried to fix it by check smoothedLevel is expire，it works for me
https://github.com/livekit/livekit/commit/0c62503f00443914f481fd06a95a9846de0c32c3

**Comment by boks1971** [comment]
Looks like there are no packets when Android mutes (possible it is happening with hardware mute from an external device?). Something like the check from @shoyu666 is needed to ensure that idle returns inactive. Will have a look.

**Comment by boks1971** [comment]
Should be addressed in the ☝️ PR. @shoyu666 thank you for the suggestion.
